src/main/java/picard/sam/markduplicates/UmiUtil.java

```diff
@@ -32,12 +35,110 @@
  * @author mduran
  */
 
-public class UmiUtil {
+class UmiUtil {
+
+    public static final String DUPLEX_UMI_DELIMITER = "-";
+    public static final String TOP_STRAND_DUPLEX = "/A";
+    public static final String BOTTOM_STRAND_DUPLEX = "/B";
+    public static final String CONTIG_SEPARATOR = ":";
+    public static final String UMI_NAME_SEPARATOR = "/";
+    static final String INFERRED_UMI_TAG = "inferredUmi";
+
+    /**
+     * Creates a top-strand normalized duplex UMI.
+     * Single stranded UMIs are by definition already top-strand normalized, they require no transformation.
+     * Duplex UMIs that come from a top strand read are also by definition, top-strand normalized.
+     * A duplex UMI from a bottom strand can be normalized to be identical to the UMI
+     * read from its corresponding top strand by swapping the content of the
+     * UMI around the "-" found in duplex UMIs.  For example, a bottom strand
+     * duplex UMI reading ATC-CGG when top-strand normalized will read CGG-ATC.
+     *
+     * @param record SAM record to retrieve UMI from.
+     * @param umiTag The tag used in the bam file that designates the UMI.
+     * @return Normalized Duplex UMI.  If the UMI isn't duplex, it returns the UMI unaltered.
+     */
+    static String getTopStrandNormalizedUmi(final SAMRecord record, final String umiTag, final boolean duplexUmi) {
+        if (umiTag == null) return null;
+
+        final String umi = record.getStringAttribute(umiTag);
 
-    /** Returns an instance of the UMI without dashes */
-    public static String getSanitizedUMI(final SAMRecord record, final String umiTag) {
-        String umi = record.getStringAttribute(umiTag);
         if (umi == null) return null;
-        return umi.replace("-", "");
+
+        if (duplexUmi) {
+            final String[] split = umi.split(DUPLEX_UMI_DELIMITER);
+            if (split.length != 2) {
+                throw new PicardException("Duplex UMIs must be of the form X-Y where X and Y are equal length UMIs, for example AT-GA.  Found UMI, " + umi);
+            }
+
+            if (isTopStrand(record)) {
+                return split[0] + DUPLEX_UMI_DELIMITER + split[1];
+            } else {
+                return split[1] + DUPLEX_UMI_DELIMITER + split[0];
+            }
+        } else {
```

Comment:
`else` unnecessary here, e.g.
```suggestion
        }
        return umi;
```

Or you could change the `if` on line 65 above to read:
```
if (umi == null || !duplexUmi) {
    return umi;
}
```